File: mobile/src/main/java/com/siliconlabs/bledemo/log/ConnectionStateChangeLog.java
package com.siliconlabs.bledemo.log;

import android.bluetooth.BluetoothGatt;
import android.bluetooth.BluetoothProfile;

import com.siliconlabs.bledemo.other.LogType;

public class ConnectionStateChangeLog extends Log {

    public ConnectionStateChangeLog(BluetoothGatt gatt, final int status, final int newState) {
        setLogTime(getTime());
        setLogInfo(gatt.getDevice().getAddress() + "(" + getDeviceName(gatt.getDevice().getName()) + "): "
                + parseNewState(newState) + ". Status: " + parseStatus(status));
        setLogType(LogType.INFO); //malo wazne
        setDeviceAddress(gatt.getDevice().getAddress());
    }

    private String parseNewState(int newState) {
        switch (newState) {
            case BluetoothProfile.STATE_DISCONNECTED:
                return "State Disconnected";
            case BluetoothProfile.STATE_CONNECTING:
                return "State Connecting";
            case BluetoothProfile.STATE_CONNECTED:
                return "Successful connect to device";
            case BluetoothProfile.STATE_DISCONNECTING:
                return "State Disconnecting";
            default:
                return String.valueOf(newState);
        }
    }

    private String parseStatus(int status) {
        if (status == BluetoothGatt.GATT_SUCCESS) {
            return "Success";
        }
        return String.valueOf(status);
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Models/Logs/ConnectionStateChangeLog.kt
package com.siliconlabs.bledemo.Browser.Models.Logs

import android.bluetooth.BluetoothGatt
import android.bluetooth.BluetoothProfile
import com.siliconlabs.bledemo.Browser.Models.LogType

class ConnectionStateChangeLog(gatt: BluetoothGatt, status: Int, newState: Int) : Log() {
    private fun parseNewState(newState: Int): String {
        return when (newState) {
            BluetoothProfile.STATE_DISCONNECTED -> "State Disconnected"
            BluetoothProfile.STATE_CONNECTING -> "State Connecting"
            BluetoothProfile.STATE_CONNECTED -> "Successful connect to device"
            BluetoothProfile.STATE_DISCONNECTING -> "State Disconnecting"
            else -> newState.toString()
        }
    }

    private fun parseStatus(status: Int): String {
        return if (status == BluetoothGatt.GATT_SUCCESS) {
            "Success"
        } else status.toString()
    }

    init {
        logTime = getTime()
        logInfo = (gatt.device.address + " (" + getDeviceName(gatt.device.name) + "): "
                + parseNewState(newState) + ". Status: " + parseStatus(status))
        logType = LogType.INFO
        deviceAddress = gatt.device.address
    }
}